### ISSUE TITLE
Pets bug after Pathfinding improvements

### ISSUE THREAD

**Issue by mpfans** [azerothcore/azerothcore-wotlk#4425]
##### CURRENT BEHAVIOUR:
The monster has to face the pet. The pet automatically looks for the back of the monster, so it turns in circles

##### EXPECTED BLIZZLIKE BEHAVIOUR:
Pets will hurt each other face to face with monsters

##### STEPS TO REPRODUCE THE PROBLEM:
1. update to https://github.com/azerothcore/azerothcore-wotlk/commit/504a788c63265a85ac730ce5dd6c5a067f25ee67
2. Take your pet to fight monsters
3. Then let the monster's hatred be on the pet

##### EXTRA NOTES:
I think this commit caused the bug
https://github.com/azerothcore/azerothcore-wotlk/commit/c8f43d85849c40e08d0da38913489426a0c35388

`Improved`` pet following movement. Also, they attack NPC from behind now. Thanks to @Footman

##### AC HASH/COMMIT:
https://github.com/azerothcore/azerothcore-wotlk/commit/504a788c63265a85ac730ce5dd6c5a067f25ee67

##### OPERATING SYSTEM:
ubuntu 18.04
##### MODULES:
none

**Comment by mpfans** [comment]
https://user-images.githubusercontent.com/18187177/106754612-25b23500-6668-11eb-8558-07afd089f2c5.mp4

**Comment by mpfans** [comment]
https://user-images.githubusercontent.com/18187177/106754653-306cca00-6668-11eb-9ffb-6af591a4dd9d.mp4

**Comment by mpfans** [comment]
https://user-images.githubusercontent.com/18187177/106754685-39f63200-6668-11eb-9b57-9138f18afaf2.mp4

**Comment by Yehonal** [comment]
@Footman

**Comment by Yehonal** [comment]
https://github.com/azerothcore/azerothcore-wotlk/pull/4220/commits/aab4f410be5c6fb39db4156902ff495c3a44d130#r569425205

**Comment by mpfans** [comment]
It's not just hunter pets, it's all pets,The first video is hunter's pet, and the second is warlock's pet

Monsters in the face of pets, will not cause any damage to pets, pets will cause damage to monsters

**Comment by FALL1N1** [comment]
I fixed it on a fork of mine by checking if the victim is a pet (e.g has aggro) - to ignore the angle of the NPC

e.g https://github.com/azerothcore/azerothcore-wotlk/commit/aab4f410be5c6fb39db4156902ff495c3a44d130#diff-37f75191f2da153de5a373250bbb996378b3666d88804b312fa1cd62f2c7fb30L603
something like this will fix it - if (_canMeleeAttack()) ->  if (_canMeleeAttack() || (isPet() && !isVictim())

**Comment by FrancescoBorzi** [comment]
@FALL1N1 thanks, can you please provide a PR?

**Comment by FALL1N1** [comment]
@FrancescoBorzi I need to verify that will indeed fix it and spend some time debugging it, sadly I can't do through the phone nor this week due to some tasks from my work, sorry. If it's not fixed within that period of time I will take it.

**Comment by Zoidwaffle** [comment]
I also noticed some weird behavior from my felguards (multiboxing warlocks) yesterday.

The felguards constantly running around a target they are fighting... Man, I kept [hearing this](https://www.youtube.com/watch?v=Zi_XLOBDo_Y) as they moonwalked all over Stratholme while murdering stuff.

I will happily test out a patch as well - with 4 felguards hacking away at once, they spin around like crazy right now (it does look kinda fun for a while though).
